src/service.js

```diff
@@ -1,56 +1,52 @@
 const axios = require('axios');
-const core = require('@actions/core');
+const logger = require('./logger');
 
 const handleChapterError = (chapter) => {
-  const failedChapterData = {
-    contentPath: JSON.parse(chapter.config.data).path,
-    status: chapter.status,
-    message: chapter.statusText,
-  };
+  const filePath = JSON.parse(chapter.config.data).path;
+  const errorMessage = chapter.data.message ? chapter.data.message : chapter.statusText;
 
-  core.error('----------- Error Processing Chapter------------');
-  core.error(`Content Path: ${failedChapterData.contentPath}`);
-  core.error(`Status code: ${failedChapterData.status}`);
-  core.error(`Error Message: ${failedChapterData.message}`);
+  return logger.error(filePath, 'chapter', chapter.status, errorMessage);
 };
 
-const handleChaptersResult = (createdChaptersResult) => {
-  const results = createdChaptersResult.reduce((resultAccumulator, currentResult) => {
+const handleChaptersResult = (createdChaptersResult) =>
+  createdChaptersResult.reduce((resultAccumulator, currentResult) => {
     const newResultAccumulator = { ...resultAccumulator };
-    if (currentResult.status === 500 || currentResult.status === 400) {
-      handleChapterError(currentResult);
-
-      newResultAccumulator.success = false;
-      return newResultAccumulator;
-    }
-
     if (currentResult.status === 200) {
       newResultAccumulator.results = [
         ...newResultAccumulator.results, currentResult.data.chapter_id,
       ];
       return newResultAccumulator;
     }
 
-    return newResultAccumulator;
-  },
-  { results: [], success: true });
+    if (currentResult.status !== 422) {
+      handleChapterError(currentResult);
+      newResultAccumulator.success = false;
+      return newResultAccumulator;
+    }
 
-  return results;
-};
+    return newResultAccumulator;
+  }, { results: [], success: true });
 
 const configHeaders = (apiKey) => ({ headers: { 'x-api-content-key': apiKey } });
 
-const createOneChapter = (apiUrl, body, config) => axios.post(apiUrl, body, config);
-
-const createChapters = async (arrayOfChapters, apiUrl, apiKey) => {
+const createChapter = (apiUrl, body, arrayOfAssets, apiKey) => {
   const headerObj = configHeaders(apiKey);
+  const bodyObj = {
+    markdown_commit_id: body.markdownCommitId,
+    markdown_content: body.contentMd,
+    yaml_commit_id: body.yamlCommitId,
+    yaml_content: body.contentYaml,
+    assets_list: arrayOfAssets,
```

Comment:
Temos que alterar o endpoint para receber os parametros em snake_case talvez valha uma tarefinha pra isso.

E já que vamos ter que syncar os nomes dos parametros no backend, seria bom também já mudar o nome do parametro de assets_list para assets_map e arrayOfAssets para assetsMap ou mapOfAssets

Reply:
Ja abri um PR lá e to fazendo essa alteração